src/components/Contexts/StepperDialogProvider.jsx

```diff
@@ -16,7 +16,19 @@ const StepperDialogProvider = ({ children }) => {
       const allCurrentStepsDefinitionsSorted = allCurrentStepsDefinitions.sort(
         (a, b) => a.stepIndex - b.stepIndex
       )
-      setAllCurrentSteps(allCurrentStepsDefinitionsSorted)
+
+      const { stepIndex: lastStepIndex } = allCurrentStepsDefinitionsSorted[
+        allCurrentStepsDefinitionsSorted.length - 1
+      ]
```

Comment:
:pick: `allCurrentStepsDefinitionsSorted.slice(-1).pop()` can also be used. It is easier to read/write if you know the syntax, but harder if not. In my case I prefer it but it's very opinionated.